src/Tizen.Applications.Common/Tizen.Applications/AppControl.cs

```diff
@@ -640,6 +652,86 @@ public static void SendTerminateRequest(AppControl terminateRequest)
             }
         }
 
+        /// <summary>
+        /// Sends the launch request asynchronously.
+        /// </summary>
+        /// <remarks>
+        /// The operation is mandatory information for the launch request.
+        /// If the operation is not specified, AppControlOperations.Default is used by default.
+        /// If the operation is AppControlOperations.Default, the application ID is mandatory to explicitly launch the application.<br/>
+        /// Since Tizen 2.4, the launch request of the service application over out of packages is restricted by the platform.
+        /// Also, implicit launch requests are NOT delivered to service applications since 2.4.
+        /// To launch a service application, an explicit launch request with the application ID given by property ApplicationId MUST be sent.
+        /// </remarks>
+        /// <param name="launchRequest">The AppControl.</param>
+        /// <param name="replyAfterLaunching">The callback function to be called when the reply is delivered.</param>
+        /// <exception cref="ArgumentException">Thrown when failed because of the argument is invalid.</exception>
+        /// <exception cref="Exceptions.AppNotFoundException">Thrown when the application to run is not found.</exception>
+        /// <exception cref="Exceptions.LaunchFailedException">Thrown when the request failed to launch the application.</exception>
+        /// <exception cref="Exceptions.LaunchRejectedException">Thrown when the launch request is rejected.</exception>
+        /// <exception cref="Exceptions.OutOfMemoryException">Thrown when the memory is insufficient.</exception>
+        /// <exception cref="Exceptions.PermissionDeniedException">Thrown when the permission is denied.</exception>
+        /// <exception cref="TimeoutException">Thrown when failed because of timeout.</exception>
+        /// <privilege>http://tizen.org/privilege/appmanager.launch</privilege>
+        /// <since_tizen> 6 </since_tizen>
+        public static Task<AppControlResult> SendLaunchRequestAsync(AppControl launchRequest, AppControlReplyCallback replyAfterLaunching)
+        {
+            var task = new TaskCompletionSource<AppControlResult>();
+            Interop.AppControl.ErrorCode err;
+
+            int resultId = 0;
+            lock (s_resultNativeCallbackMaps)
+            {
+                resultId = s_resultNativeCallbackId++;
+                s_resultNativeCallbackMaps[resultId] = (IntPtr handle, int result, IntPtr userData) =>
+                {
+                    task.SetResult((AppControlResult)result);
+                    s_resultNativeCallbackMaps.Remove((int)userData);
```

Comment:
you pass a replyId  into user_data
(see 701)
and you pass IntPtr.Zero into user_data
(see 705)

Did you test this code?

Reply:
in this case use local delegate
```c#
var resultCb = (IntPtr handle, int result, IntPtr userData) => {
task.SetResult((AppControlResult)result);
};
```

Reply:
I made a sample app and tested it separately. There was a mistake in moving the code.
When I checked with your suggested code, a crash occurred. The local delegate has been deleted by the garbage collector. Hmm.. According to the guide you wrote in confluence, the delegate is managed as a dictionary object. I think it's right.